contracts/mocks/MockValidator.sol

```diff
@@ -28,11 +29,13 @@ contract MockValidator is IValidator {
     }
 
     function onInstall(bytes calldata data) external {
+        require(IModuleManager(msg.sender).isModuleInstalled(MODULE_TYPE_VALIDATOR, address(this), ""), "Validator is still installed");
         smartAccountOwners[msg.sender] = address(bytes20(data));
     }
 
     function onUninstall(bytes calldata data) external {
         data;
+        require(!IModuleManager(msg.sender).isModuleInstalled(MODULE_TYPE_VALIDATOR, address(this), ""), "Validator is still installed");
```

Comment:
same I feel for onUninstall 
how much gas does this action increase?
this will also affect smart account deployment cost.

Reply:
for stateful validator it would always be referred by msg.sender address